flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/HMM.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.classification
+
+import org.apache.flink.api.scala.DataSet
+
+import org.apache.flink.api.scala._
+import org.apache.flink.ml.math.{DenseMatrix, DenseVector}
+import org.apache.flink.ml.optimization.BaumWelch
+import org.apache.flink.util.Collector
+import org.slf4j.LoggerFactory
+
+class HMM {
```

Comment:
Do you have a reference for your implementation? If so, would be good to link to the paper. Also some documentation of `HMM` is missing. See other FlinkML algorithms how they do it.